src/psyclone/configuration.py

```diff
@@ -777,30 +777,24 @@ def create_dict_from_list(input_list):
         return return_dict
 
     @staticmethod
-    def create_numeric_dict_from_list(input_list):
-
+    def get_precision_map_dict(section):
+        '''Extracts the precision map values from the psyclone.cfg file
+        and converts them to a dictionary with integer values.
+        :returns: The precision maps to be used by this API.
+        :rtype: Dictionary of integers
+        '''
+        precisions_list = section.getlist("precision_map")
         return_dict = {}
-        for entry in input_list:
-            try:
-                key, value = entry.split(":")[:2]
-            except ValueError as err:
-                # Raised when split does not return two elements:
-                raise ConfigurationError(
-                    f"Invalid format for mapping: '{entry.strip()}'") from err
-            # Try to convert to float or int
-            try:
-                if isinstance(value, str):
-                    if '.' in value:
-                        value = int(float(value))
-                    else:
-                        value = int(value)
-            except ValueError as err:
-                # Raised when split does not return two elements:
+        return_dict = APISpecificConfig.create_dict_from_list(precisions_list)
+
+        for key, value in return_dict.items():
+            if value.isdigit():
```

Comment:
`isdigit` is a good idea but I see that it also accepts unicode characters:
![image](https://github.com/stfc/PSyclone/assets/17409212/e246f8d5-9ee7-4cfb-b776-93a1792337d8)
I think just asking Python to convert it to an integer within a try...except is the way to go.

Reply:
That will mean we will also need to check that the resulting integer value is `>0`.

Reply:
I chose to use both [`.isdecimal()`](https://www.programiz.com/python-programming/methods/string/isdecimal) _and_ `.isdigit()` which successfully filters out the special unicode characters, negative integers, floats, or strings containing letters in one line.

Reply:
Great, thanks.